types/index.d.ts

```diff
@@ -229,6 +229,27 @@ declare namespace jest {
      */
     toBeObject(): R;
 
+    /**
+     * Use `.toChange` when checking if a value has changed.
+     * @example
+     * expect(() => value--).toChange(() => value);
+     */
+    toChange(checker: () => number): R;
+
+    /**
+     * Use `.toChangeTo` when checking if a value changed to a specific value.
+     * @example
+     * expect(() => Model.deleteAll()).toChangeTo(() => Model.count(), 0);
+     */
+    toChangeTo<Val>(checker: () => Val, to?: Val): R;
```

Comment:
`to` shouldn't be optional, right?

Also probably should be `<E = unknown>` instead of `<Val>`.